stl/inc/istream

```diff
@@ -619,14 +619,12 @@ public:
             return -1;
         }
 
-        bool _Sync_failed = true; // sync fails if an exception is thrown
         _TRY_IO_BEGIN
-        _Sync_failed = _Rdbuf->pubsync() == -1;
-        _CATCH_IO_END
-        if (_Sync_failed) {
+        if (_Rdbuf->pubsync() == -1) {
             _Myios::setstate(ios_base::badbit);
             return -1;
         }
+        _CATCH_IO_END

```

Comment:
Note, as far as I can tell this is equivalent.

I assume the old code was there before the `_CATCH_IO_END` macro came around

Reply:
My understanding of iostreams is extremely low, but it seems to me that `_CATCH_IO_END` doesn't always rethrow. If you follow the function calls you will get to `ios_base::clear` and see that `_RERAISE` only happens if `_State` satisfies the `_Except` mask which could be set to 0, in which case no exception would be thrown. In such case the current implementation would return -1 but after this change it would return 0 (meaning success).

Reply:
That is a great ... *catch*

Thanks, fixed and added tests for the return value